lib/internal/child_process.js

```diff
@@ -478,14 +497,24 @@ function setupChannel(target, channel) {
     });
   });
 
-  target.send = function(message, handle) {
-    if (!this.connected)
-      this.emit('error', new Error('channel closed'));
-    else
-      this._send(message, handle, false);
+  target.send = function(message, handle, callback) {
+    if (typeof handle === 'function') {
+      callback = handle;
+      handle = undefined;
+    }
+    if (this.connected) {
+      this._send(message, handle, false, callback);
+      return;
+    }
+    const ex = new Error('channel closed');
+    if (typeof callback === 'function') {
+      process.nextTick(function() { callback(ex); });
+    } else {
+      this.emit('error', ex);  // FIXME(bnoordhuis) Defer to next tick.
```

Comment:
After looking through the code base, there are actually many places where errors are emitted on the same tick. Maybe the `FIXME` should just be removed instead.